index.js

```diff
@@ -386,6 +410,122 @@ exports.getLastThreadPost = function (thread) {
   return msg
 }
 
+
+/* post revision utils */
+
+exports.getRevisions = function(ssb, thread, callback) {
+  function collectRevisions(thread, callback) {
+    // this function walks the revisions of a given thread, collecting them up
+    // asyncly
+    var deepThread = pluckRecursively(thread, 'related')[0]
+
+    var threadRevisionLogCB = multicb({pluck: 1})
+    deepThread
+    .filter(function(relatedMsg) { return isaRevisionTo(relatedMsg, thread) })
+    .forEach(function(edit) {
+      exports.createRevisionLog(ssb, edit, threadRevisionLogCB())
+    })
+    
+    threadRevisionLogCB(function(err, revLogs) {
+      if (err) callback(err)
+      const connectedLogs = revLogs
+        .filter(function(revLog) { 
+          // filter down to the log that links to this thread, by finding the log
+          // whose messages point to this thread as root 
+          // we should only need to test one message if createRevisionLog is
+          // passing tests
+          return revLog[revLog.length-1].key === thread.key
+        })
+
+      // the longest such log is the one that contains all of the revisions
+      const logLengths = connectedLogs.map(function(log) { return log.length })
+      const thisLog = connectedLogs.find(function(log) {
+                        return log.length === Math.max.apply(Math, logLengths)
+                      })
+      callback(null, thisLog)
+    })
+    
+  }
+            
+
+  if (!thread.hasOwnProperty('related')) { 
+    // if the thread doesn't have its related objects,fetch them
+    ssb.relatedMessages({id: thread.key, count: true}, function(err, enrichedThread) {
+      if (err) callback (err)
+      // if still no related objects
+      else if (!enrichedThread.hasOwnProperty('related')) callback(null, [])
+      else collectRevisions(enrichedThread, callback)
+    })
+  } else {
+    collectRevisions(thread, callback)
+  }  
+}
+
+exports.getLatestRevision = function (ssb, thread, callback) {  
+  var msg = thread
+  if (!msg.hasOwnProperty('related')) {
+    ssb.relatedMessages({id: msg.key, count: true}, function(err, richMsg) {
+      if (err) callback (err)
+      // if still no related objects
+      else if (!richMsg.hasOwnProperty('related')) { callback(null, richMsg) }
+      else { traverseRelatedEdits(richMsg, callback) }
+    })
+  } else if (!msg.hasOwnProperty('related')) {
+    callback(null, msg)
+  } else {
+    traverseRelatedEdits(msg, callback)
+  }
+}
+
+exports.createRevisionLog = function (ssb, msg, callback) {
```

Comment:
I dont think this method is needed in getRevisions, and Im not sure it's needed at all.

This method is basically a copy of `fetchThreadRootID`, which mainly exists to deal with threads that are malformed (a message with a bad/missing `root`). I dont think we need that kind of tolerance in the revisions, so I think this function isnt needed.